### ISSUE TITLE
Cannot create a schema with key "g", type "Map" and alias string starting with char "g".

### ISSUE THREAD

**Issue by chrischen** [Automattic/mongoose#7464]
**Do you want to request a *feature* or report a *bug*?**
BUG

**What is the current behavior?**
`Error: Virtual path "gString" conflicts with a real path in the schema`

**If the current behavior is a bug, please provide the steps to reproduce.**
Create a schema like so:

```
new Schema({
  t: { alias: 'type', type: String },
  g: { alias: "gString", type: Map, of: Number }
});
```
**What is the expected behavior?**
It should work with no errors.

**Please mention your node.js, mongoose and MongoDB version.**
Mongo 3.6.2, node v9.10.1, mongoose 5.4.7.

**Comment by mhombach** [comment]
Can you test my script and tell me if this throws an error on your machine (you have to change the db-settings in the first 4 lines)? Because it doesn't throw on mine.

```js
const mongoose = require('mongoose');
const DB = 'test_db';
const URI = `mongodb://localhost:27017/${DB}`;
const OPTS = { family: 4, useNewUrlParser: true };

let mySchema = new mongoose.Schema({
  t: { alias: 'type', type: String },
  g: { alias: "gString", type: Map, of: Number }
});

var myModel = mongoose.model('my', mySchema);

async function run() {
  await mongoose.connect(URI, OPTS);
  var test1 = new myModel();
  test1.type = "test";
  test1.gString = { a: 101, b: 102};
  var test2 = await test1.save();
  console.log("test2 = " + JSON.stringify(test2));
}
run();
```

**Comment by noo-zh** [comment]
I have the same issue

```
const schema = new Schema({
  d: {
    type: Map,
    of: Schema.Types.Mixed,
    alias: 'details'
  }
});
```

_(of course, my model is more complex)_

I am getting the same error `Virtual path "details" conflicts with a real path in the schema` since I upgraded from 5.3.16 to 5.4.12. Maybe something is wrong with 5.4.*...? When I remove "alias" field, it starts working. But that's not the case I need, so I did have to downgrade back.
